internal/rbd/controllerserver.go

```diff
@@ -508,6 +508,15 @@ func (cs *ControllerServer) repairExistingVolume(ctx context.Context, req *csi.C
 
 			return nil, err
 		}
+
+	default:
+		// setup encryption again to make sure everything is in place.
```

Comment:
Question:
* This is only for PVC. What about the PVC Clone and PVC Snapshot?
* What about the fileEncryption?

Reply:
> Question:
> 
> * This is only for PVC. What about the PVC Clone and PVC Snapshot?
 
For clone
https://github.com/ceph/ceph-csi/blob/07e9dede2cb10ceaa055b9b2486f688edf9008d6/internal/rbd/rbd_journal.go#L337
For snapshot, just few ones above to copy encryption config 


> * What about the fileEncryption?

This pr handles for block encryption,  
We can handle the same for file encryption in another pr.
@irq0 can you please take a look at this?

Reply:
Open an issue so that we dont forget it?